gnd/src/main/java/com/google/android/gnd/repository/ProjectRepository.java

```diff
@@ -64,9 +64,11 @@ public ProjectRepository(
     this.localValueStore = localValueStore;
 
     // BehaviorProcessor re-emits last requested project id to late subscribers.
+    // TODO (DT) - Why is this necessary? Under what conditions would there be late subscribers?
```

Comment:
"Late subscribers" here just means UIs that initialized after the active project was loaded. Since the order of initialization of views in Android as often hard to mentally model, we treat it as non-deterministic (and perhaps sometimes it is?). 

I agree Rx concepts are not sufficiently self-describing; that's been a pain point in onboarding. I'm not sure what the best way to mitigate this is; what level of Rx knowledge do you think out docs should assume to strike the right balance between verbosity/redundancy and clarity?

Reply:
I think a class/architecture diagram would reduce the onboarding barrier significantly. I've been sketching one out and will publish when I have it in a useful digital form. 

It's definitely not necessary to explain Rx concepts, but may be necessary to explain which Rx concepts Ground is using and why it's using them.

Reply:
+1. @mmisim is this something you might be able to help with, i.e. a basic architecture doc -> entry in our Wiki?

Reply:
Yes, I can help with this. I think starting with an architecture diagram and fleshing out into a wiki article would be a good approach. @dturner — do you want me to take a look a what you have so far?